library/appointments.inc.php

```diff
@@ -371,9 +376,17 @@ function fetchAppointments($from_date, $to_date, $patient_id = null, $provider_i
 
     $where = "";
 
+ //RM multiple providers
     if ($provider_id) {
-        $where .= " AND e.pc_aid = ?";
-        array_push($sqlBindArray, $provider_id);
+        $quantity = sizeof($provider_id) ;
```

Comment:
Going to step through the code but this breaks the patient flow board report when searching on a specific provider.  Not sure why but will investigate.

Reply:
Looks like it fails when $provider_id is a string when the function expects an array. 
Following code fixes it:
```
    if (!empty($provider_id)) {
        // provider_id can be a string or an array
        if (is_array($provider_id)) {
            $quantity = sizeof($provider_id);
            $where .= " AND ( e.pc_aid = ?";
            for ($i = 1; $i < $quantity; $i++) {
                $where .= " OR e.pc_aid = ? ";
            }
            $where .= ")";
            foreach ($provider_id as $x) {
                array_push($sqlBindArray, $x);
            }
        } else {
            $where .= " AND e.pc_aid = ?";
            array_push($sqlBindArray, $provider_id);
        }
    }
```

Reply:
oh dear, thanks, i thought i'd tested that !!

Reply:
You gave me access to push so I went ahead and fixed it.

Reply:
Everything else checks out good.  Once the CI engine passes I can bring it in, unless you want to review my changes.

Reply:
many thanks, no i trust you completely!! please do bring it in. I'm just cooking but will read the solution more carefully later.